r/tests/testthat/test-dplyr-funcs-datetime.R

```diff
@@ -2638,14 +2694,18 @@ test_that("build_formats() and build_format_from_order()", {
     )
   )
 
-  # ab not supported yet
-  expect_error(
-    build_formats("abd"),
-    '"abd" `orders` not supported in Arrow'
+  expect_equal(
+    build_format_from_order("abp"),
+    c("%a", "%b","%p", "%a", "%b", "%p")
+  )
+
+  expect_equal(
+    build_format_from_order("uYM"),
+    c("%y-%M", "%Y-%M", "%y%M", "%Y%M")
```

Comment:
These are odd. My suspicion is something is not quite right here: https://github.com/apache/arrow/pull/13506/files#diff-8fdef8da9950901a3c1535c97b0370aad5a4fced647e29c76c22bc6114cf4880R222

Reply:
The processing of `order` happens in `build_formats`, https://github.com/apache/arrow/blob/ef6049a2ee5673d0944a0b4f70ff9c70e0419a22/r/R/dplyr-datetime-helpers.R#L166-L167, so that orders getting to `build_format_from_order()` are "clean".

Reply:
For example, due to line 166, no order should have a `%` inside.